lib/srv/uacc/uacc.h

```diff
@@ -65,12 +63,18 @@ static int check_abs_path_err(const char* buffer) {
             case ENOTDIR: return UACC_UTMP_FAILED_OPEN;
             default: return UACC_UTMP_OTHER_ERROR;
         }
+}
+
+static int check_abs_path_err(const char* buffer) {
+    // check for errors
+    if (buffer == NULL) {
+        return status_from_errno();
     }
 
     // check for GNU extension errors
     if (errno == EACCES || errno == ENOENT) {
-        UACC_PATH_ERR = (char*)malloc(PATH_MAX);
-        strcpy(UACC_PATH_ERR, buffer);
+        uaccPathErr = (char*)malloc(PATH_MAX);
+        strcpy(uaccPathErr, buffer);
```

Comment:
Although very unlikely, `malloc` above can fail.

I know this was the original design, but the memory management of `uaccPathErr` makes me nervous. I have carefully reviewed how it is assigned and freed, and while I don't think there is an error _now_, I still wish we had used a simpler pattern.

Perhaps we can use a static buffer instead? This should be safe given that we lock `accountDb` anyway.

Reply:
Malloc in CGO will just abort on failure I'm pretty sure so this should be a non-concern.

Reply:
Global/statics scare me and can go wrong fast. Could we also use the relatively simple pattern of passing in pointer to    an array on the stack from go?